src/Compilers/CSharp/Portable/LanguageVersion.cs

```diff
@@ -130,9 +125,25 @@ public enum LanguageVersion
         CSharp8 = 800,
 
         /// <summary>
-        /// The latest version of the language supported.
+        /// The latest major supported version.
+        /// </summary>
+        LatestMajor = int.MaxValue - 2,
+
+        /// <summary>
+        /// The latest supported version of the language.
+        /// </summary>
+        Latest = int.MaxValue - 1,
+
+        /// <summary>
+        /// Preview of the next language version.
         /// </summary>
-        Latest = int.MaxValue,
```

Comment:
This is a breaking change, right? If I recall, enum values are burned into an assembly, so code built against Latest will be considerd as "Preview" (which is appropirate perhaps?) but then upgrading to newer packages and rebuilding causes your burned in value to change?

Reply:
That's true, but the change to Default is also a breaking change, and I see no clear way to avoid that, since we're explicitly changing the meaning of Default in this version for the command-line compiler.

Reply:
Sure, the reinterpretation of default is expected. But you're creating additional breaks which don't seem necessary to me and seem more worrisome. Specifically, a tool creating a Compilation that compiled against "Latest" but running against newer assemblies (consider this happens in Visual Studio!) is now running against "Preview" which is absolutely not what it meant. I get it's "nice" to have the ordering be LatestMajor, Latest, Preview for the sentinel values, but it's causing what seems to me to be avoidable breaks.

I'm also not sure if there's analyzer breaks here too, because what happens if an analyzer wrote "GetEffectiveVersion(LanguageVersion.Latest)"? That'll return preview too.

Reply:
Makes sense. I'll keep MaxValue as Latest and move the others to smaller values.